02_ColorTableViewTask/ViewController.swift

```diff
@@ -34,16 +31,16 @@ class ViewController: UIViewController {
     @IBOutlet weak var editButton: UIButton!
     
     private var colors: [ColorDataSource] = [
-        ColorDataSource(color: UIColor(named: "Deep Teal"), name: "Deep Teal"),
-        ColorDataSource(color: UIColor(named: "Catalina Blue"), name: "Catalina Blue"),
-        ColorDataSource(color: UIColor(named: "Dark Indigo"), name: "Dark Indigo"),
-        ColorDataSource(color: UIColor(named: "Ripe Plum"), name: "Ripe Plum"),
-        ColorDataSource(color: UIColor(named: "Mulberry Wood"), name: "Mulberry Wood"),
-        ColorDataSource(color: UIColor(named: "Kenyan Copper"), name: "Kenyan Copper"),
-        ColorDataSource(color: UIColor(named: "Chestnut"), name: "Chestnut"),
-        ColorDataSource(color: UIColor(named: "Antique Bronze"), name: "Antique Bronze")
+        ColorDataSource(colorName: "Deep Teal", describtion: "Deep Teal describtion"),
+        ColorDataSource(colorName: "Catalina Blue", describtion: "Catalina Blue describtion"),
+        ColorDataSource(colorName: "Dark Indigo", describtion: "Dark Indigo describtion"),
+        ColorDataSource(colorName: "Ripe Plum", describtion: "Ripe Plum describtion"),
+        ColorDataSource(colorName: "Mulberry Wood", describtion: "Mulberry Wood describtion"),
+        ColorDataSource(colorName: "Kenyan Copper", describtion: "Kenyan Copper describtion"),
+        ColorDataSource(colorName: "Chestnut", describtion: "Chestnut describtion"),
+        ColorDataSource(colorName: "Antique Bronze", describtion: "Antique Bronze describtion")
     ]
-    var colorsId: [Int] = [0, 1, 2, 3, 4, 5, 6, 7]
+    var colorsId: [Int] = []
```

Comment:
I'm not sure if this should be optional instead of an empty array. The colorsId will be initialized in viewDidLoad anyway. In case, it is the first time launching the app, it will be initialized using getColorsId, other than that it will use the array in user defaults.